### PR TITLE
Add hacktoberfest-related documentation

### PR DESCRIPTION
**Pull request by mpenkov**
I'm following up on hacktoberfest hints from the numfocus people. This document should help people get started contributing for Hacktoberfest. I've already gone through the tickets for gensim and smart_open and picked out several dozen for people to work on.

Once this is merged, it'd be good to attract people via Twitter, etc.

### CODE REVIEW

**Comment by mpenkov** [comment]
@piskvorky @gojomo Can you please have a look at this? I'm going for balance between:

1. Attracting new contributors
2. Discouraging spammy contributions

File: HACKTOBERFEST.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# :pizza: Hacktoberfest 2019 :beer:
```
Comment (menshikh-iv) [inline HACKTOBERFEST.md]: Big link from the top of readme (image maybe?) to this document will be useful for new contributors

File: HACKTOBERFEST.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+It's that time of the year again!
+[Hacktoberfest](https://hacktoberfest.digitalocean.com) is here, and `gensim` needs **your** help.
+We've prepared a list of good issues to work on [here](https://github.com/RaRe-Technologies/gensim/labels/hacktoberfest).
+
+If the learning curve for `gensim` is a bit steep, give the [smart_open](https://github.com/RaRe-Technologies/smart_open) repository a try.
+`smart_open` is an important dependency of `gensim`: it performs file I/O over a variety of protocols and formats.
+There's also a list of Hacktoberfest-friendly issues to work on [here](https://github.com/RaRe-Technologies/smart_open/labels/hacktoberfest).
+
+Of course, we welcome contributions on any of the existing issues, not just the ones labeled `hacktoberfest`.
```
Comment (menshikh-iv) [inline HACKTOBERFEST.md]: add  link to search by "hacktoberfest" label

File: HACKTOBERFEST.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Please avoid making innocuous changes without sufficient motivation (e.g. changing code formatting, etc).
+
+## Before Contributing
+
+Check out the following:
+
+- [First-time contributors guide](https://github.com/firstcontributions/first-contributions): if this is your first time contributing on GitHub.
+- [Hacktoberfest rules](https://hacktoberfest.digitalocean.com/faq#rules): read this in full
+- [Developer page](https://github.com/RaRe-Technologies/gensim/wiki/Developer-page) on our Wiki: for the git flow, code style, etc.
+
```
Comment (menshikh-iv) [inline HACKTOBERFEST.md]: add links to the person that contributor can ask about anything (if any questions)

File: HACKTOBERFEST.md (hunk 1 above)
Comment (mpenkov) [reply HACKTOBERFEST.md]: Good idea, done

File: HACKTOBERFEST.md (hunk 3 above)
Comment (mpenkov) [reply HACKTOBERFEST.md]: OK, linked to my own Telegram and Twitter

**Comment by piskvorky** [comment]
Great idea! 👍

File: HACKTOBERFEST.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+It's that time of the year again!
+[Hacktoberfest](https://hacktoberfest.digitalocean.com) is here, and `gensim` needs **your** help.
+We've prepared a list of good issues to work on: [gensim hacktoberfest issues](https://github.com/RaRe-Technologies/gensim/labels/hacktoberfest).
+
+If the learning curve for `gensim` is a bit steep, give the [smart_open](https://github.com/RaRe-Technologies/smart_open) repository a try.
+`smart_open` is an important dependency of `gensim`: it performs file I/O over a variety of protocols and formats.
+There's also a list of Hacktoberfest-friendly issues to work on: [smart_open hacktoberfest issues](https://github.com/RaRe-Technologies/smart_open/labels/hacktoberfest).
+
+Of course, we welcome contributions on any of the existing issues, not just the ones labeled `hacktoberfest`.
+Just be sure that no-one else is already working on it first.
```
Comment (gojomo) [inline HACKTOBERFEST.md]: This is a little vague, and may result in people either (1) hesitating to make a small fix (by feeling they must ask, "is anyone working on this?" first), or (2) prematurely claiming an issue without follow-through (by declaring "I'm working on this" then disappearing for an arbitrary amount of time). 

Perhaps instead: 

> If an issue is simple & quick, you can just submit your PR, with a proper reference to the issue it addresses. 
> 
> If an issue requires a little more work, but you have a good idea of how to proceed & know when you'll be submitting some initial work, please post a short note about your plans to the issue, or a "work-in-progress" ("[WIP]") pull-request indicating work is underway, to help avoid wasted duplicate work.

File: HACKTOBERFEST.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+Furthermore, we also welcome contributions not connected to an existing issue.
+This includes things like fixing typos in documentation, docstrings, etc.
+If you make such contributions, please make the motivation behind the contribution clear.
```
Comment (gojomo) [inline HACKTOBERFEST.md]: May be worth noting: "You could start such a contribution with a new pull-request, or if you think it requires other discussion beforehand, as a separate new issue."

**Comment by gojomo** [comment]
Looks good!

Might we also want to guide people to either the Google Groups forum, or Gitter chat, for other more general questions not related to any specific issue?

**Comment by mpenkov** [comment]
Thanks for the comments everyone.

@piskvorky Can you please tweet and link to https://github.com/RaRe-Technologies/gensim/blob/develop/HACKTOBERFEST.md?

**Pull request merged**